cmd/kubeless/function/deploy.go

```diff
@@ -177,44 +183,52 @@ var deployCmd = &cobra.Command{
 		}
 
 		f, err := getFunctionDescription(funcName, ns, handler, file, funcDeps, runtime, runtimeImage, mem, cpu, timeout, imagePullPolicy, port, headless, envs, labels, secrets, defaultFunctionSpec)
-
 		if err != nil {
 			logrus.Fatal(err)
 		}
 
-		kubelessClient, err := utils.GetKubelessClientOutCluster()
-		if err != nil {
-			logrus.Fatal(err)
-		}
-
-		logrus.Infof("Deploying function...")
-		err = utils.CreateFunctionCustomResource(kubelessClient, f)
-		if err != nil {
-			logrus.Fatalf("Failed to deploy %s. Received:\n%s", funcName, err)
-		}
-		logrus.Infof("Function %s submitted for deployment", funcName)
-		logrus.Infof("Check the deployment status executing 'kubeless function ls %s'", funcName)
-
-		if schedule != "" {
-			cronJobTrigger := kubelessApi.CronJobTrigger{}
-			cronJobTrigger.TypeMeta = metav1.TypeMeta{
-				Kind:       "CronJobTrigger",
-				APIVersion: "kubeless.io/v1beta1",
-			}
-			cronJobTrigger.ObjectMeta = metav1.ObjectMeta{
-				Name:      funcName,
-				Namespace: ns,
+		if dryrun == true {
+			b, err := json.MarshalIndent(f, "", "    ")
+			if err != nil {
+				logrus.Fatal(err)
 			}
-			cronJobTrigger.ObjectMeta.Labels = map[string]string{
-				"created-by": "kubeless",
-				"function":   funcName,
+			fmt.Println(string(b[:]))
```

Comment:
I would `return` here so you don't need to write the rest of the code in an `else` block